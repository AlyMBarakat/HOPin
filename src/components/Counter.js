import React, { useState } from 'react';
import {
    Text,
    View,
    StyleSheet,
    TouchableOpacity,

} from 'react-native';

const Counter = ({ name }) => {
    const [count, setCount] = useState(0);

    const handleCount = (operation) => {
        if (operation === "+" && count < 4)
            setCount(count => count + 1);
        else if (operation === "-" && count > 0)
            setCount(count => count - 1);
        return
    }

    return (
        <View style={{ marginBottom: 22, }}>
            <Text style={styles.inputHeader}>{name}</Text>
            <View style={styles.input}>
                <Text style={{ color: "#2D3436", fontSize: 20, width: 20 }}>{count}</Text>
                <View style={{ flexDirection: 'row' }}>
                    <TouchableOpacity
                        style={styles.button}
                        onPress={() => handleCount("-")}
                    >
                        <Text style={{ color: "#fff", fontSize: 20, fontWeight: 'bold' }}>-</Text>
                    </TouchableOpacity>
                    <TouchableOpacity
                        style={styles.button}
                        onPress={() => handleCount("+")}
                    >
                        <Text style={{ color: "#fff", fontSize: 20, fontWeight: 'bold' }}>+</Text>
                    </TouchableOpacity>
                </View>
            </View>
        </View>
    );
};


const styles = StyleSheet.create({
    inputHeader: {
        color: "#fff",
        fontSize: 15,
        fontWeight: 'bold',
        alignSelf: 'flex-start',
        marginBottom: 12,
    },
    input: {
        backgroundColor: '#72B5A8',
        borderWidth: 3,
        borderRadius: 5,
        borderColor: '#3C796D',
        paddingHorizontal: 6,
        paddingVertical: 10,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        alignSelf: 'flex-start',
    },
    button: {
        backgroundColor: '#3C796D',
        borderRadius: 5,
        paddingHorizontal: 10,
        paddingVertical: 5,
        marginLeft: 10,
    }
})

export default Counter;
