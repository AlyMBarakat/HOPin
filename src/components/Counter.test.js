import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './Counter';

test('renders Counter correctly', () => {
  const tree = renderer.create(<Counter name="counter" />).toJSON();
  expect(tree).toMatchSnapshot();
});
