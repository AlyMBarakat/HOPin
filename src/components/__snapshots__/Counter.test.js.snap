// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Counter correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 22,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "#fff",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginBottom": 12,
      }
    }
  >
    counter
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#72B5A8",
        "borderColor": "#3C796D",
        "borderRadius": 5,
        "borderWidth": 3,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 6,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2D3436",
          "fontSize": 20,
          "width": 20,
        }
      }
    >
      0
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#3C796D",
            "borderRadius": 5,
            "marginLeft": 10,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          -
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#3C796D",
            "borderRadius": 5,
            "marginLeft": 10,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
</View>
`;
